src/items/associated-items.md

```diff
@@ -109,13 +109,15 @@ following types:
 - [`Arc<Self>`]
 - [`Pin<P>`] where `P` is one of the above types except `Self`.
 
-The `Self` term can be replaced with the type being implemented.
+The `Self` term can be replaced with the type being implemented, including
+type aliases for the type, or any nested combination of the above types.
```

Comment:
Maybe add a note re. object safety (when it's not)?

Reply:
I added more info in the object safety section.  The intent is that it can all be rewritten when Receiver is stabilized.

I also added some examples. Do you think they are useful? I'm concerned they take up a lot of space.  The rules are fairly concise, but I find it hard to visualize how they translate to code without some examples.

Reply:
It's probably fine :)